metals/src/main/scala/scala/meta/internal/implementation/ImplementationProvider.scala

```diff
@@ -35,7 +37,7 @@ final class ImplementationProvider(
     buildTargets: BuildTargets,
     buffer: Buffers,
     definitionProvider: DefinitionProvider
-) {
+)(implicit ec: ExecutionContext) {
```

Comment:
will limit the thread size?

Reply:
It's using`Executors.newCachedThreadPool()`, so it tries to reuse existing threads, but it doesn't limit them.

Reply:
@olafurpg what do you think? Should we maybe add a thread limit on the ThreadPool ?

Reply:
I'm concerned about submitting so many runnables to the Metals server execution context. I would prefer to use parallel collections `.foreach` for this use case, collecting results in a `j.u.c.ConcurrentLinkedQueue`. I think parallel collections use their own execution context which is by default suitable for CPU bound work. I don't have a good experience using non-`foreach` combinators in parallel collections.

Reply:
Changed to parallel collections, thanks!